### PR TITLE
[Integration][Sentry] - Add Users and Teams

### PR DESCRIPTION
**Pull request by PeyGis**
# Description

What - Added sentry users and teams

Why - For improved data model

How - Added sentryUser and sentryTeam blueprint. And added a relation between issue and user

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

Screenshot 2024-12-12 at 7 40 42 PM

Screenshot 2024-12-12 at 7 41 03 PM

## API Documentation

[Users](https://docs.sentry.io/api/organizations/list-an-organizations-members/)
[Teams](https://docs.sentry.io/api/teams/list-an-organizations-teams/)
[Team Members](https://docs.sentry.io/api/teams/list-a-teams-members/)

### CODE REVIEW

**Review by matan84 — CHANGES_REQUESTED** [review]
some minor comments

File: integrations/sentry/clients/sentry.py [hunk 1]
```diff
@@ -242,3 +242,31 @@ async def get_projects_tags_from_projects(
...
+    async def get_paginated_teams(
+        self,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for teams in self._get_paginated_resource(
+            f"{self.api_url}/organizations/{self.organization}/teams/"
+        ):
+            yield teams
+
+    @cache_iterator_result()
+    async def get_paginated_users(
```
[the remark this answers is not in the document]
Comment (PeyGis) [reply integrations/sentry/clients/sentry.py:256]: we already log how many members are in each team
        logger.info(f"Received {len(team_members_List)} members for team {team_slug}")

File: integrations/sentry/.port/resources/blueprints.json [hunk 2]
```diff
@@ -1,4 +1,103 @@
...
+          "icon": "DefaultProperty",
+          "title": "Organization Role",
+          "type": "string",
+          "enum": [
+            "member",
+            "admin",
+            "owner",
+            "manager",
+            "biling"
+          ],
```
Comment (Tankilevitch) [inline integrations/sentry/.port/resources/blueprints.json:37]: is that something static by sentry or can be adjusted by each organization?

File: integrations/sentry/clients/sentry.py [hunk 3]
```diff
@@ -242,3 +242,36 @@ async def get_projects_tags_from_projects(
...
+    @cache_iterator_result()
+    async def get_paginated_teams(
+        self,
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for teams in self._get_paginated_resource(
+            f"{self.api_url}/organizations/{self.organization}/teams/"
+        ):
+            yield teams
+
+    @cache_iterator_result()
```
Comment (Tankilevitch) [inline integrations/sentry/clients/sentry.py:255]: why you need to cache this?

File: integrations/sentry/clients/sentry.py [hunk 4]
```diff
@@ -242,3 +242,36 @@ async def get_projects_tags_from_projects(
             if project_tags_batch:
                 project_tags.append(project_tags_batch)
         return flatten_list(project_tags)
+
+    @cache_iterator_result()
```
Comment (Tankilevitch) [inline integrations/sentry/clients/sentry.py:246]: same here?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: integrations/sentry/.port/resources/blueprints.json (hunk 2 above)
Comment (PeyGis) [reply integrations/sentry/.port/resources/blueprints.json:37]: this is static from the API and consistent with the Sentry UI

File: integrations/sentry/main.py [hunk 5]
```diff
@@ -25,6 +30,45 @@ def init_client() -> SentryClient:
     return sentry_client
 
 
+async def enrich_team_with_members(
+    sentry_client: SentryClient,
+    team_batch: list[dict[str, Any]],
+) -> list[dict[str, Any]]:
+    async def fetch_team_members(team_slug: str) -> list[dict[str, Any]]:
+        return await sentry_client.get_team_members(team_slug)
+
```
[the remark this answers is not in the document]
Comment (PeyGis) [reply integrations/sentry/main.py:39]: Can you rephrase? I didn't understand the question

[the remark this answers is not in the document]
Comment (Tankilevitch) [reply integrations/sentry/main.py:39]: feels redundant method I think

[the remark this answers is not in the document]
Comment (Tankilevitch) [reply integrations/sentry/main.py:39]: it doesn't add anything specific

**Review by Tankilevitch — APPROVED** [review]
LGTM

**Review by matan84 — APPROVED** [review]
LGTM
